logprep/processor/dissector/rule.py

```diff
@@ -84,25 +89,41 @@
 
 START = r"%\{"
 END = r"\}"
-VALID_TARGET_FIELD = r"[^\}\%\{\}\+]*"
+VALID_TARGET_FIELD = r"[^\}\%\{\}\+\/\|]*"
 APPEND_WITH_SEPERATOR = r"(\+\([^%]+\))"
 APPEND_WITHOUT_SEPERATOR = r"(\+(?!\([^%]))"
 INDIRECT_FIELD_NOTATION = r"([&\?]))"
 VALID_ACTION = rf"({APPEND_WITH_SEPERATOR}|{APPEND_WITHOUT_SEPERATOR}|{INDIRECT_FIELD_NOTATION}"
 VALID_POSITION = r"(\/\d*)"
-DISSECT = rf"{START}{VALID_ACTION}?{VALID_TARGET_FIELD}{VALID_POSITION}?{END}"
+VALID_DATATYPE = r"(\|(int|float|str|bool))"
+POSITION_OR_DATATYPE = rf"({VALID_POSITION}|{VALID_DATATYPE})"
+DISSECT = rf"{START}{VALID_ACTION}?{VALID_TARGET_FIELD}{POSITION_OR_DATATYPE}?{END}"
 DELIMETER = r"([^%]+)"
 ACTION = r"(?P<action>[+])?"
 SEPERATOR = r"(\((?P<separator>.+)\))?"
-TARGET_FIELD = r"(?P<target_field>[^\/]*)"
+TARGET_FIELD = r"(?P<target_field>[^\/\|]*)"
 POSITION = r"(\/(?P<position>\d*))?"
-SECTION_MATCH = rf"{START}{ACTION}{SEPERATOR}{TARGET_FIELD}{POSITION}{END}(?P<delimeter>.*)"
+DATATYPE = r"(\|(?P<datatype>int|float|bool))?"
+SECTION_MATCH = (
+    rf"{START}{ACTION}{SEPERATOR}{TARGET_FIELD}{POSITION}{DATATYPE}{END}(?P<delimeter>.*)"
+)
 
 
 def _do_nothing(*_):
     return
 
 
+def str_to_bool(input_str: str) -> bool:
+    """converts an input string to bool by meaning"""
+    if input_str.lower() in ("yes", "true", "on", "y"):
```

Comment:
I would add a check for None to ensure that the object input_str is not None before calling lower

Reply:
indeed, but I resolved it by asking for forgiveness, because the case should occur rarely often.